src/bananaInBoxRule.ts

```diff
@@ -0,0 +1,76 @@
+import * as Lint from 'tslint';
+import * as ts from 'typescript';
+import * as ast from '@angular/compiler';
+import {BasicTemplateAstVisitor} from './angular/templates/basicTemplateAstVisitor';
+import {NgWalker} from './angular/ngWalker';
+
+
+const BoxInABananaOpen = '([';
+const BoxInABananaClose = '])';
+const BananaInABoxOpen = '[(';
+const BananaInABoxClose = ')]';
+const BoxInABananaRe = new RegExp('\\(\\[(.*?)\\]\\)(.*?)');
+
+const getReplacements = (text: ast.BoundEventAst, absolutePosition: number) => {
+  const expr: string = (text.sourceSpan as any).toString();
+  const internalStart = expr.indexOf('([');
+  const internalEnd = expr.lastIndexOf('])');
+  const len = internalEnd - internalStart - BoxInABananaClose.length;
+  const trimmed = expr.substr(internalStart + BoxInABananaOpen.length, len).trim();
+
+  return [
+    new Lint.Replacement(absolutePosition,
+      internalEnd - internalStart + BoxInABananaClose.length,
+      `${BananaInABoxOpen}${trimmed}${BananaInABoxClose}`)
+  ];
+};
+
+class BananaInBoxTemplateVisitor extends BasicTemplateAstVisitor {
+
+  visitEvent(prop: ast.BoundEventAst, context: any): any {
+
+    if (prop.sourceSpan) {
+      // Note that will not be reliable for different interpolation symbols
+      let error = null;
+      const expr: any = (<any>prop.sourceSpan).toString();
+        if (BoxInABananaRe.test(expr)) {
+        error = 'The box is in a banana! Expecting Banana in a box [(expr)]';
```

Comment:
This warning message is fun but lets change it to something like: "Invalid binding syntax. Use [(expr)] instead".